dvc/repo/tree.py

```diff
@@ -437,6 +481,27 @@ def copytree(self, top, dest):
                 with self.open(src, mode="rb") as fobj:
                     copy_fobj_to_file(fobj, dest_dir / fname)
 
-    @property
-    def hash_jobs(self):  # pylint: disable=invalid-overridden-method
-        return self.repo.tree.hash_jobs
+    @cached_property
+    def hash_jobs(self):
+        return self.tree.hash_jobs
+
+    def in_subtree(self, path):
+        return self._find_subtree(path)
+
+    @post_processing("\r\n".join)
+    def _visualize(self, top, **kwargs):
+        """`tree`-like output, useful for debugging/visualizing"""
+        indent = 4
+        spacing = " " * indent
+        tee = "├── "
+        last = "└── "
+        for root, _, files in self.walk(top, **kwargs):
+            level = root.replace(top, "").count(os.sep)
+            indent = spacing * level
+            yield "{}{}/".format(indent, os.path.basename(root))
+            sub_indent = spacing * (level + 1)
+            length = len(files)
+            for i, f in enumerate(files):
+                yield "{}{}{}".format(
+                    sub_indent, tee if i + 1 != length else last, f
+                )
```

Comment:
+1, do you intend to keep it? Looks like debugging feature.

Reply:
yes, I'd like to keep this. :smile: